### DISCUSSION TITLE
When opening the terminal in Theia, it becomes unresponsive (hangs).

### DISCUSSION THREAD

**Discussion by GN0816** [eclipse-theia/theia#14619] [Q&A]
When starting a Theia container using Docker, I encounter an issue where the spawn-helper process consumes 100% CPU when opening a terminal in the browser. As a result, the Theia page in the browser shows an "offline" status, although the Docker container is still running normally.
When Theia calls the terminal, there is a process called spawn-helper. When using strace -p to monitor this process, it continuously outputs close(42189951) = -1 EBADF (Bad file descriptor).
![1](https://github.com/user-attachments/assets/cd38f91f-a7b4-4965-8316-540d9f37e969)

**Comment by msujew** [comment]
Hey @GN0816,

we are aware of the issue, see https://github.com/eclipse-theia/theia/pull/14342 which attempts to fix this. However, due to many breaking (behavioral) changes in node-pty, the library we use for spawning terminal processes, we were unable to proceed on the PR so far.
